EIPS/eip-3675.md

```diff
@@ -0,0 +1,244 @@
+
+Beginning with the first `POS_BLOCK_FINALIZED` event, the networking stack **MUST** remove the handlers corresponding to the following ETH protocol messages:
+* [`NewBlockHashes (0x01)`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblockhashes-0x01)
+* [`NewBlock (0x07)`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblock-0x07)
+
+
+## Rationale
+
+The changes specified in this EIP target a minimal requisite set of consensus and client software modifications to safely replace the existing proof-of-work consensus algorithm with the new proof-of-stake consensus represented by the already in-production beacon chain.
+
+Minimalism in this changeset was chosen to reduce the complexity of the unprecedented task of hot-swapping the live consensus of the Ethereum network in order both to increase  the safety of the operation and to reduce the time to production. Additionally, a minimal changeset helps ensure that *most* smart contracts and services will continue to function as intended during and after the transition with little to no required intervention.
+
+### Total difficulty triggering the upgrade
+
+See [Security considerations](#Terminal-total-difficulty-vs-block-number).
+
+### Halting the import of PoW blocks
+
+See [Security considerations](#Halt-the-importing-of-PoW-blocks).
+
+### Replacing block fields with constants
+
+Deprecated block fields are replaced with constant values to ensure the block format remains backwards compatible. Preserving the block format aids existing smart contracts and services in providing uninterrupted service during and after this transition.
+
+This change introduces an additional validity rule that enforces the replacement of deprecated block fields.
+
+### Replacing `difficulty` with `0`
+
+After deprecating the proof-of-work the notion of difficulty no longer exists and replacing the block header **`difficulty`** field with `0` constant is semantically sound.
+
+### Deprecating `extraData`
+
+The **`extraData`** field is deprecated in favour of the corresponding **`graffiti`** field of the beacon block.
+
+### Changing block validity rules
+
+The rule set enforcing the PoW seal validity is replaced with the corresponding PoS rules along with the consensus upgrade as the rationale behind this change.
+
+An additional rule validating a set of deprecated block fields is required by the block format changes introduced by this specification.
+
+### Removing block rewards
+
+Existing rewards for producing and sealing blocks are deprecated along with the PoW mechanism. The new PoS consensus becomes both responsible for sealing blocks and for issuing block rewards once this specification enters into effect.
+
+### Supplanting fork choice rule
+
+The fork choice rule of the PoW mechanism becomes completely irrelevant after the upgrade and is replaced with the corresponding rule of the new PoS consensus mechanism.
+
+### Removing block gossip
```

Comment:
I think there should be a mention that there should still be tx gossip to avoid thinking that we should remove gossip on eth1 altogether

Reply:
Added a note that the logic of other message handlers must remain unchanged